packages/cli/lib/LocalDevManagerV2.js

```diff
@@ -31,14 +38,51 @@ class LocalDevManagerV2 {
   }
 
   async start() {
-    console.clear();
     SpinniesManager.removeAll();
     SpinniesManager.init();
 
+    const componentsByType = await findProjectComponents(this.projectSourceDir);
+
+    const componentTypes = Object.keys(componentsByType);
+
+    if (!componentTypes.length) {
+      logger.log();
+      logger.error(i18n(`${i18nKey}.noComponents`));
+      process.exit(EXIT_CODES.SUCCESS);
+    }
+
+    const runnableComponentsByType = componentTypes.reduce((acc, type) => {
```

Comment:
`runnableComponentsByType` is a subset of the `componentsByType` object. It only includes components that have `runnable` set to true. For apps, this means it only includes apps that contain react extensions.